app/controllers/application_controller.rb

```diff
@@ -1,7 +1,17 @@
-class ApplicationController < ActionController::API
+class ApplicationController < ActionController::Base
   include GDS::SSO::ControllerMethods
 
   before_action :require_signin_permission!
 
-  protect_from_forgery with: :exception
+  rescue_from ActionController::ParameterMissing do |e|
+    render json: { error: { message: e.message } }, status: 400
+  end
+
+  rescue_from ActiveRecord::RecordNotFound do |e|
+    render json: { error: { message: e.message } }, status: 404
+  end
+
+  def payload
+    @payload ||= JSON.parse(request.body.read).deep_symbolize_keys
+  end
```

Comment:
This looks familiar :wink: - I don't think the precedent set by Publishing API is a great one to follow though as it fights against rails conventions

if, on a post request, we have a header of content-type application/json Rails should automatically set the params to be json parsed, it should also be a [HashWithIndifferentAccess](http://api.rubyonrails.org/v4.2/classes/ActiveSupport/HashWithIndifferentAccess.html) so wouldn't need the deep symbolize either